File: option.c
#include <stdio.h>
#include <time.h>

void sort(int scores[], int size);
void printarray(int scores[], int size, char str[]);

void sort(int scores[], int size) {
    int cnt;
    for (int i = 0; i < size; i++) {
        for (int j = i + 1; j < size; j++) {
            if (scores[i] < scores[j]) {
                cnt = scores[i];
                scores[i] = scores[j];
                scores[j] = cnt;
            }
        }
    }
}

void printarray(int scores[], int size, char str[]) {
    printf("%s = ", str);
    for (int i = 0; i < size; i++) {
        printf("%d ", scores[i]);
    }
    printf("\n");
}

int main() {
    const int SIZE = 5; //定数として配列のサイズを宣言                          
    int scores[SIZE] = {0, 60, 70, 100, 90};
    clock_t start_time, end_time;

    // 処理開始の時間を記録
    start_time = clock();

    // 処理コード
    printarray(scores, SIZE, "scores");
    sort(scores, SIZE);
    printarray(scores, SIZE, "results");

    // 処理終了の時間を記録
    end_time = clock();

    // CPU時間を計算
    long cpu_time = (long)(end_time - start_time);

    /* 秒に直す */
    double sec = (double)cpu_time / CLOCKS_PER_SEC;

    printf("処理時間 : %f 秒\n", sec);

    return 0;
}

File: option.go
package main

import (
	"fmt"
	"time"
)

func main() {
	scores := [5]int{0, 60, 70, 100, 90}
	// 計測開始
	start := time.Now()

	printarray(scores[:], len(scores), "scores")
	sort(scores[:], len(scores))
	printarray(scores[:], len(scores), "results")

	end := time.Now()
	// 経過時間を計算し、秒に変換
	elapsedTime := end.Sub(start).Seconds()
	// 経過時間を出力
	fmt.Printf("処理時間 : %f 秒\n", elapsedTime)
}

func printarray(scores []int, size int, str string) {
	fmt.Printf("%s = ", str)
	for i := 0; i < size; i++ {
		fmt.Printf("%d ", scores[i])
	}
	fmt.Printf("\n")
}

func sort(scores []int, size int) {
	for i := 0; i < size; i++ {
		for j := i + 1; j < size; j++ {
			if scores[i] < scores[j] {
				cnt := scores[i]
				scores[i] = scores[j]
				scores[j] = cnt
			}
		}
	}
}
